lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1248,20 +1247,30 @@ static void copyParameterArgumentsForApply(
 /// recursively converts the new function just like how the old function is
 /// converted. If the new function's generic signature is specified, it is used
 /// to create substitution maps for reapplied `partial_apply` instructions.
-static SILValue
-reapplyFunctionConversion(
-    SILValue newFunc, SILValue oldFunc, SILValue oldConvertedFunc,
-    SILBuilder &builder, SILLocation loc,
+static SILValue reapplyFunctionConversion(
+    ADContext &context, SILValue newFunc, SILValue oldFunc,
+    SILValue oldConvertedFunc, SILBuilder &builder, SILLocation loc,
     SmallVectorImpl<AllocStackInst *> &newBuffersToDealloc,
-    GenericSignature newFuncGenSig = GenericSignature()) {
+    GenericSignature newFuncGenSig = GenericSignature(),
+    IndexSubset *reabstractionThunkIndices = nullptr) {
   // If the old func is the new func, then there's no conversion.
   if (oldFunc == oldConvertedFunc)
     return newFunc;
   // Handle a few instruction cases.
+  // copy_value
+  if (auto *cvi = dyn_cast<CopyValueInst>(oldConvertedFunc)) {
+    auto innerNewFunc = reapplyFunctionConversion(
+        context, newFunc, oldFunc, cvi->getOperand(), builder, loc,
+        newBuffersToDealloc, newFuncGenSig);
```

Comment:
do the `reabstractionThunkIndices` need to be passed into the recursive call here?

similar comment about all the recursive calls.

Reply:
This hasn't been necessary in practice but seems good for robustness, e.g. to handle function conversion of the result of reabstraction thunk `partial_apply`s. I'll update this soon.

Reply:
Done in https://github.com/apple/swift/pull/28570/commits/6eb7a070fd6734c1deb1220556038e4267d349ef.